GVFS/GVFS.Common/PlaceholderListDatabase.cs

```diff
@@ -89,7 +102,14 @@ public void RemoveAndFlush(string path)
             }
         }
 
-        public List<PlaceholderData> GetAllEntries()
+        /// <summary>
+        /// Gets all entries and intializes placeholderChangesWhileRebuildingList in preparation fr a call to WriteAllEntriesAndFlush.

```

Comment:
What you're describing here is internal implementation details of this class, not really relevant to the caller. What's relevant to the caller is: if you call this method and don't call `WriteAllEntriesAndFlush`, you can lose data.